core/src/main/scala/kafka/server/RemoteLeaderEndPoint.scala

```diff
@@ -224,8 +222,10 @@ class RemoteLeaderEndPoint(logPrefix: String,
         metadataVersion.fetchRequestVersion
       }
       val requestBuilder = if (readOnlyTopics.isEmpty) {
+        // regular replication: use replica fetch request
         FetchRequest.Builder.forReplica(version, brokerConfig.brokerId, brokerEpochSupplier(), maxWait, minBytes, fetchData.toSend)
       } else {
+        // cluster linking: use consumer fetch request to avoid consistency checks
```

Comment:
I think the main reason we want to use consumer fetch now is to allow fetch till LSO, though that be changed later. So `to avoid consistency checks` might not be true.